Mathlib/RingTheory/Algebraic/MvPolynomial.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2019 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+import Mathlib.Algebra.MvPolynomial.Supported
+import Mathlib.RingTheory.Algebraic.Basic
+
+/-!
+# Transcendental elements in `MvPolynomial`
+
+This file lists some results on some elements in `MvPolynomial σ R` being transcendental
+over the base ring `R` and subrings `MvPolynomial.supported` of `MvPolynomial σ R`.
+-/
+
+universe u v w
+
+open Polynomial
+
+namespace MvPolynomial
+
+variable {σ : Type*} (R : Type*) [CommRing R]
+
+-- TODO: move to suitable place
+private theorem rename_polynomial_aeval_X
+    {σ τ R : Type*} [CommSemiring R] (f : σ → τ) (i : σ) (p : R[X]) :
+    rename f (Polynomial.aeval (X i) p) = Polynomial.aeval (X (f i) : MvPolynomial τ R) p := by
+  rw [← AlgHom.comp_apply]
+  congr 1; ext1; simp
```

Comment:
#find_home says this can be moved to Mathlib.Algebra.MvPolynomial.Equiv.

Reply:
Great observation! The result could be proved in one line `rw [← AlgHom.comp_apply, ← Polynomial.aeval_algHom, rename_X]`. I'm thinking

- if this should be a simp lemma,
- or, since this result is only used once, can it be inlined to the proof below

Reply:
It feels like a good enough `@[simp]` lemma, so let me upstream it to `Mathlib.Algebra.MvPolynomial.Equiv` with your suggested proof.

Reply:
Maybe a good follow-up project would be to give a name to the embedding from `Polynomial` to `MvPolynomial` given by `aeval (X i) p`.

Reply:
> Maybe a good follow-up project would be to give a name to the embedding from `Polynomial` to `MvPolynomial` given by `aeval (X i) p`.

Good idea! I haven't think of it yet.